rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -503,48 +502,70 @@ protected void registerPegIn(
         );
 
         PeginProcessAction peginProcessAction = peginEvaluationResult.getPeginProcessAction();
-        if (peginProcessAction == PeginProcessAction.CAN_BE_REGISTERED) {
-            logger.debug("[{}] Peg-in is valid, going to register", METHOD_NAME);
-            executePegIn(btcTx, peginInformation, totalAmount);
-        } else {
-            Optional<RejectedPeginReason> rejectedPeginReasonOptional = peginEvaluationResult.getRejectedPeginReason();
-            if (rejectedPeginReasonOptional.isEmpty()) {
-                // This flow should never be reached. There should always be a rejected pegin reason.
-                String message = "Invalid state. No rejected reason was returned from evaluatePegin method";
-                logger.error("[{}}] {}", METHOD_NAME, message);
-                throw new IllegalStateException(message);
-            }
 
-            RejectedPeginReason rejectedPeginReason = rejectedPeginReasonOptional.get();
-            logger.debug("[{}] Rejected peg-in, reason {}", METHOD_NAME, rejectedPeginReason);
-            eventLogger.logRejectedPegin(btcTx, rejectedPeginReason);
-            if (peginProcessAction == PeginProcessAction.CAN_BE_REFUNDED) {
-                logger.debug("[{}] Refunding to address {} ", METHOD_NAME, peginInformation.getBtcRefundAddress());
-                generateRejectionRelease(btcTx, peginInformation.getBtcRefundAddress(), rskTxHash, totalAmount);
+        switch (peginProcessAction) {
+            case REGISTER -> {
+                logger.debug("[{}] Peg-in is valid, going to register", METHOD_NAME);
+                executePegIn(btcTx, peginInformation, totalAmount);
+            }
+            case REFUND -> {
+                peginEvaluationResult.getRejectedPeginReason()
+                    .map(reason -> {
+                        logger.debug("[{}] Rejected peg-in, reason {}", METHOD_NAME, reason);
+                        eventLogger.logRejectedPegin(btcTx, reason);
+                        return reason;
```

Comment:
Do you need this return here? Nobody is storing that value

Reply:
You right. It can be removed.